### ISSUE TITLE
Need to index existence of tar ball even if it cannot be indexed properly.

### ISSUE THREAD

**Issue by ndokos** [distributed-system-analysis/pbench#1352]
ATM, we give up completely on tar balls that don't meet some minimal criteria (e.g. existence and goodness of metadata.log file, good start run date, good hostnames in sosreports etc). But we want to use ES as the authority for the data that are available, not just the data that are good enough to be indexed. So we need to modify pbench-index.py to index at least a minimal run document that will allow us to get at the tarball, even if it fails to index properly.

**Comment by riya-17** [comment]
Isn't this issue has already been addressed?

**Comment by dbutenhof** [comment]
No, it hasn't; we do the tarball consistency checks and throw an exception without doing any indexing if it doesn't pass.

We do create a `Dataset` ... but so far the data queries don't report a dataset that has a `Dataset` but no Elasticsearch presence. We _could_ change that, e.g. by doing a `Dataset` query for controller/name match that's in `QUARANTINE` state (which such a dataset would be) in addition to the Elasticsearch query. I'd considered using my `Dataset` DB table to look up controllers in the first place rather than Elasticsearch, when I added users; I didn't mostly because we use ES because of the fast replicated index and search, and bypassing that felt like going in the wrong direction.

But a more direct interpretation of this issue would be that we create an Elasticsearch `run-data` document, at least, even if we quarantine the tarball because it's "bad". That suggests doing the consistency checks in stages, and erroring out only when we get to an indexing stage we can't handle (for example, pulling out the iteration data)... but presumably returning a new error counter for "partially indexed" tarballs which would be reported and rolled into the `server-report` document.
